zkevm-circuits/src/tx_circuit.rs

```diff
@@ -128,6 +128,208 @@ struct RlpTableInputValue<F: Field> {
+                    (AccessListAddressesLen, AccessListStorageKeysLen, 1, 1, 0),
+                    (AccessListStorageKeysLen, AccessListRLC, 1, 1, 0),
+                    (AccessListRLC, MaxFeePerGas, 1, 1, 0),
+                    (MaxFeePerGas, MaxPriorityFeePerGas, 1, 1, 0),
+                    (MaxPriorityFeePerGas, BlockNumber, 1, 1, 0),
+                    // Transition into dynamic section of tx_table
+                    (BlockNumber, Nonce, 0, 1, 0),
+                    (BlockNumber, CallData, 1, 1, 1),
+                    (BlockNumber, CallData, 0, 1, 1),
+                    (BlockNumber, TxFieldTag::AccessListAddress, 1, 1, 1),
+                    (BlockNumber, TxFieldTag::AccessListAddress, 0, 1, 1),
+                    // Transition between dynamic tags of tx_table
+                    (CallData, CallData, 1, 0, 0),
+                    (CallData, CallData, 0, 1, 0),
+                    (CallData, TxFieldTag::AccessListAddress, 1, 1, 0),
+                    (CallData, TxFieldTag::AccessListAddress, 0, 1, 0),
+                    (
+                        TxFieldTag::AccessListAddress,
+                        TxFieldTag::AccessListAddress,
+                        1,
+                        0,
+                        0,
+                    ),
+                    (
+                        TxFieldTag::AccessListAddress,
+                        TxFieldTag::AccessListAddress,
+                        0,
+                        1,
+                        0,
+                    ),
+                    (
+                        TxFieldTag::AccessListAddress,
+                        TxFieldTag::AccessListStorageKey,
+                        1,
+                        0,
+                        0,
+                    ),
+                    (
+                        TxFieldTag::AccessListStorageKey,
+                        TxFieldTag::AccessListStorageKey,
+                        1,
+                        0,
+                        0,
+                    ),
+                    (
+                        TxFieldTag::AccessListStorageKey,
+                        TxFieldTag::AccessListAddress,
+                        1,
+                        0,
+                        0,
+                    ),
+                    (
+                        TxFieldTag::AccessListStorageKey,
+                        TxFieldTag::AccessListAddress,
+                        0,
+                        1,
+                        0,
+                    ),
+                    (TxFieldTag::AccessListAddress, CallData, 0, 1, 0),
+                    (TxFieldTag::AccessListStorageKey, CallData, 0, 1, 0),
+                    // Continue padding. Padding has the Calldata tag
+                    (CallData, CallData, 1, 1, 0),
+                    // Precompile call transition scenarios
+                    (CallData, TxFieldTag::Null, 1, 1, 0),
+                    (CallData, TxFieldTag::Null, 0, 1, 0),
```

Comment:
why these two transitions?

Reply:
Since transitions `CallData -> Null` and `Null -> Nonce` are allowed, a malicious prover can include two different transactions that have same tx_id in the tx table.

Reply:
By adding the additional condition on the tag transition lookup, the above two scenarios are no longer needed.